src/clarity.rs

```diff
@@ -344,19 +345,18 @@ pub fn invoke_command(invoked_by: &str, args: &[String]) {
             };
 
             let mut outer_sp = db_conn.begin_save_point_raw();
+            let contract_analysis: ContractAnalysis;
 
             { 
                 let mut analysis_db = AnalysisDatabase::from_savepoint(
                     friendly_expect(outer_sp.savepoint(),
                                     "Failed to initialize savepoint for analysis"));
                 let mut ast = friendly_expect(parse(&contract_content), "Failed to parse program.");
-
-                friendly_expect(type_check(contract_name, &mut ast, &mut analysis_db, true),
-                                "Type check error.");
-
+                let contract_analysis_result = type_check(contract_name, &mut ast, &mut analysis_db, true);
```

Comment:
This is going to conflict with @lgalabru's diagnostics PR.  Just a heads-up.

Reply:
Merged in dev branch, good to go!